cmake/Helpers.cmake

```diff
@@ -255,3 +255,25 @@ function(check_python_module module)
         "Note: You need to have pip installed for this Python version.")
   endif()
 endfunction()
+
+macro(find_supported_python_version)
+  if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.19")
+    find_package(Python 3.6...<3.10.999 COMPONENTS Interpreter REQUIRED)
+  else()
+    # The version range syntax is only supported from CMake 3.19 on.
+    # So, for previous versions, we manually search for an allowed python version
+    foreach (python_version 3.10 3.9 3.8 3.7 3.6)
+      find_package(Python ${python_version} EXACT COMPONENTS Interpreter)
```

Comment:
Wouldn't `EXACT` here only match version 3.10 but not 3.10.1?

Reply:
`EXACT` will match every 3.10.x version.

Reply:
I tried without the `EXACT`, but in this case, cmake interprets this version number as the minimum required. On MacOS, `find_package(Python 3.10)` loaded Python 3.11, because that was the latest one installed in the machine.